### ISSUE TITLE
Rule Change: (no-unused-vars)

### ISSUE THREAD

**Issue by roryabraham** [eslint/eslint#16487]
### What rule do you want to change?

no-unused-vars

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
function myUsedFunction() {
    console.log('foo');
}

function myUnusedFunction() {
    console.log('bar');
}

export {
    // This function is imported and used by other modules
    myUsedFunction,

    // This function is not used by other modules
    myUnusedFunction,
};
```

### What does the rule currently do for this code?

No flag – the above code passes.

### What will the rule do after it's changed?

The rule should flag variables that are exported but never imported. Or if the variables are imported like so:

```js
import * as MyLib from './MyLib';
```

But `MyLib.myUnusedFunction` is never used, then the rule should produce a warning regarding this unused variable.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @roryabraham, thanks for the issue!

Core rules do not track variable usage in other modules.

You can use the `import/no-unused-modules` plugin rule with option `unusedExports: true` to catch variables that are exported but never imported.

https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-unused-modules.md

**Comment by roryabraham** [comment]
Thanks for the tip, @mdjermanovic!
